pkg/scheduler/reconciliation/inmemory.go

```diff
@@ -213,6 +213,10 @@ func (r *InMemoryReconciliationRepository) UpdateOperationState(schedulingID, co
 		return &repository.EntityNotFoundError{}
 	}
 
+	if err := operationAlreadyInState(op, state); err != nil {
```

Comment:
It turned out that we have also cases in which an update of an operation is required even if it's already in state: the heartbeat messages are continuously updating an operation (even when its state hasn't changed).

Maybe we add a flag to the `UpdateOperationState` to allow updates even if the state hasn't changed. The only case in which the update ins't allowed would be when a worker tries to pick an operation. Otherwise, updating exists states would be acceptable.

Reply:
Done